step/src/main/java/com/google/sps/User.java

```diff
@@ -16,59 +16,65 @@
 
 import java.util.Collection;
 import java.util.HashSet;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 
 public class User {
   
-  private long id;
+  private String id;
   private String email;
   private String name;
   private Collection<User> matches;
+  
 
-  // TODO: determine what fields we want.
-  // TODO: update the constructor so that it only needs to take
-  //       in an id, and the other fields can then be looked up
-  //       in the database. this update will be made when data-
-  //       -store is integrated
-  public User(long id, String email, String name) {
+  public User(String id) {
     this.id = id;
```

Comment:
Seems like it would be easier to have the User class determine the id rather than having it as a parameter.

Reply:
Not entirely sure I get what you mean by this. Do you mean to not take in an id and instead make a call to UserService for the id within the class?

Reply:
Oh is this the id provided by userService.getCurrentUser().getUserId()? If so, then I get what you're doing here but I think a more intuitive variable name would be something like userServiceUserId so it's more specific. I was under the impression we were still creating our own user ids.